### PR TITLE
[PRNG] Add check to PRNG to make sure that unsigned integer arithmetic is wrapping

### PR DESCRIPTION
**Pull request by tkonolige**
This PR adds a simple test to `threefry_generate` to make sure that unsigned integer arithmetic is wrapping. We have no guarantee of wrapping on our platforms, so this is the best we can do. I've added the test to `threefry_generate` as it should be called before other threefry functions and it should called infrequently.

@altanh @antinucleon @junrushao1994 @eric-haibin-lin @MarisaKirisame

### CODE REVIEW

**Comment by tqchen** [comment]
@tkonolige One thing to note that assert may not be available in every platform, better to put down a comment. Might be useful to have it as a separate question.

File: python/tvm/topi/random/kernel.py [hunk 1]
```diff
@@ -20,6 +20,8 @@
 from ... import tir
 from ...tir import ir_builder
 
+import numpy as np
```
Comment (altanh) [inline python/tvm/topi/random/kernel.py:23]: probably better if we move the test function definition to the `test_topi_prng.py` file also

**Review by altanh — CHANGES_REQUESTED** [review]
LGTM, I'd prefer if we could have the test function defined in the test file since it doesn't depend on any TOPI PRNG internals but not a huge deal

File: python/tvm/topi/random/kernel.py (hunk 1 above)
Comment (tkonolige) [reply python/tvm/topi/random/kernel.py:23]: I want people to be able to run this function themselves to check if their platform is supported. If I put it in the test file, you can't import and use it. You'd have to edit the test manually to include your platform.

Comment (altanh) [reply python/tvm/topi/random/kernel.py:23]: I see, hmm. That makes sense, and I'm not sure there's a better way to indicate to devs other than the warning comment, so looks good. Can you fix the lint error? something about import numpy ordering

File: python/tvm/topi/random/kernel.py [hunk 2]
```diff
@@ -234,6 +242,18 @@ def gen_ir(gen_ptr, out_gen_ptr, out_array_ptr):
         out_gen = irb.buffer_ptr(out_gen_ptr)
         out_array = irb.buffer_ptr(out_array_ptr)
 
+        # Check that unsigned arithmetic wraps, as it is required to implement threefry correctly.
+        irb.emit(
```
Comment (altanh) [inline python/tvm/topi/random/kernel.py:246]: echoing Tianqi's comment, will compilation fail on a target that doesn't support `AssertStmt` but would in theory have everything needed for Threefry PRNG?

Comment (tkonolige) [reply python/tvm/topi/random/kernel.py:246]: @tqchen If the platform does not support assert, does it fail to compile or does it just not include the assert?

Comment (tqchen) [reply python/tvm/topi/random/kernel.py:246]: The behavior can be backend dependent, I think right now the assert will be omitted

Comment (tkonolige) [reply python/tvm/topi/random/kernel.py:246]: That should be fine then.

**Review by altanh — APPROVED** [review]

**Comment by altanh** [comment]
bump @tqchen @ZihengJiang 

I think this is good to go since we resolved the assert discussion

**Review by ZihengJiang — APPROVED** [review]

**Comment by ZihengJiang** [comment]
Merged. Thanks @tkonolige @altanh

**Pull request merged**